src/main/java/org/dcsa/ebl/service/impl/TransportDocumentTOServiceImpl.java

```diff
@@ -1,28 +1,207 @@
+        List<ChargeTO> chargeTOs = transportDocumentTO.getCharges();
+        if (chargeTOs == null || chargeTOs.isEmpty()) {
+            transportDocumentTO.setCharges(Collections.emptyList());
+            return Flux.empty();
+        } else {
+            List<Charge> charges = new ArrayList<>(chargeTOs.size());
+            chargeTOs.stream().forEach(chargeTO -> {
+                // Insert TransportDocumentID on all Charges
+                chargeTO.setTransportDocumentID(transportDocumentTO.getId());
+                // Create a Charge object for all ChargeTOs
+                Charge charge = MappingUtil.instanceFrom(
+                        chargeTO,
+                        Charge::new,
+                        AbstractCharge.class
+                );
+                charges.add(charge);
+            });
+
+            // Save all Charges in one bulk
+            return chargeService.createAll(charges);
+        }
+    }
+
+    private Flux<Clause> createClauses(TransportDocumentTO transportDocumentTO) {
+        List<Clause> clauses = transportDocumentTO.getClauses();
+        if (clauses == null || clauses.isEmpty()) {
+            transportDocumentTO.setClauses(Collections.emptyList());
+            return Flux.empty();
+        } else {
+            // Save all Clauses in one Bulk
+            return clauseService.createAll(clauses)
+                    .concatMap(clause ->
+                            // Make sure many-many relations are created
+                            clauseService.createTransportDocumentClauseRelation(clause.getId(), transportDocumentTO.getId())
+                                    .thenReturn(clause)
+                    );
+        }
     }
 
+    private Mono<TransportPlan> createTransportPlan(TransportDocumentTO transportDocumentTO) {
+        // TODO: ...
+        return Mono.just(new TransportPlan());
+    }
+
+    @Transactional
     @Override
-    public Class<TransportDocumentTO> getModelClass() {
-        return TransportDocumentTO.class;
+    public Mono<TransportDocumentTO> findById(UUID transportDocumentID, boolean includeCharges) {
+        TransportDocumentTO transportDocumentTO = new TransportDocumentTO();
+
+        return Flux.concat(
+                transportDocumentService.findById(transportDocumentID)
+                    .flatMap(
+                            transportDocument -> {
+                                MappingUtil.copyFields(
+                                        transportDocument,
+                                        transportDocumentTO,
+                                        AbstractTransportDocument.class
+                                );
+                                if (transportDocument.getShippingInstructionID() == null) {
+                                    return Mono.error(new GetException("No ShippingInstruction connected to this TransportDocument - internal error!"));
```

Comment:
Please do not use `GetException` for internal errors as it translates to  "400 Bad Request" (`IllegalStateException` or so is fine and gives us "500 Internal Server Error")